server/application/application.go

```diff
@@ -388,6 +388,7 @@ func (s *Server) Watch(q *ApplicationQuery, ws ApplicationService_WatchServer) e
 	case <-ws.Context().Done():
 		w.Stop()
 	case <-done:
+		w.Stop()
```

Comment:
Why have `w.Stop()` in two places and not make it a deferred call right after successfully creating the watch?

Reply:
First stop is required to stop `for next := range w.ResultChan()` loop after context got canceled.  Second is required to make sure that stop is called after function completes. I agree that it is better to use defer this second case. Done.